cmd/collector/app/span_processor.go

```diff
@@ -177,3 +209,79 @@ func (sp *spanProcessor) enqueueSpan(span *model.Span, originalFormat SpanFormat
 	}
 	return sp.queue.Produce(item)
 }
+
+func (sp *spanProcessor) StartStateReporting(reportPeriod time.Duration) {
+	sp.background(reportPeriod, sp.updateGauges)
+}
+
+func (sp *spanProcessor) StartDynQueueSizeUpdater(reportPeriod time.Duration) {
+	sp.background(reportPeriod, sp.updateQueueSize)
+}
+
+func (sp *spanProcessor) background(reportPeriod time.Duration, callback func()) {
+	go func() {
+		ticker := time.NewTicker(reportPeriod)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				callback()
+			case <-sp.stopCh:
+				return
+			}
+		}
+	}()
+}
+
+func (sp *spanProcessor) updateQueueSize() {
+	if sp.spansProcessed.Load() == 0 {
+		return
+	}
```

Comment:
this check is redundant, it is covered by the condition in L250

Reply:
+1